packages/vm/src/runTx.ts

```diff
@@ -619,6 +673,19 @@ async function _runTx(this: VM, opts: RunTxOpts): Promise<RunTxResult> {
     }
   }
 
+  /**
+   * Cleanup code of accounts written to in a 7702 transaction
+   */
+
+  for (const str of writtenAddresses) {
+    const address = Address.fromString(str)
+    await this.stateManager.putContractCode(address, new Uint8Array())
+    // TODO verify if this is necessary
```

Comment:
I think we absolutely have to do this.  Otherwise the `codeHash` won't be null and we can/will have other downstream issues if somebody else tries to `getContractCode`, they will get a non `null` and `KECCAK_NULL` value for an account that shouldn't have code.

Reply:
In my test cases, if I remove the `modify` stuff, it still passes. But I cannot think of a scenario where it would not pass - but I could imagine if there is a complex scenario then there might be an edge case where we need it :thinking:

Reply:
if we have a extcodehash for EOA in a following transaction of th, its likely it will get the old codeshash.
should be possible to construct a testcase